src/views/SettingsPage.vue

```diff
@@ -68,27 +66,19 @@ import {
   personCircleOutline,
   settingsOutline,
 } from "ionicons/icons";
-import { ref } from "vue";
+import { onMounted, ref, watch } from "vue";
+import { STORAGE_KEYS } from "../utils/constant";
+import { updateTheme } from "../utils/theme";
 
-const selectValue = ref(localStorage.getItem("theme") ?? "system");
+const themeType = ref(
+  localStorage.getItem(STORAGE_KEYS.THEME_TYPE) ?? "system"
+);
+watch(themeType, () => {
+  localStorage.setItem(STORAGE_KEYS.THEME_TYPE, themeType.value);
+  updateTheme();
+});
 
-const prefersDark = window.matchMedia("(prefers-color-scheme: dark)");
-
-const toggleDarkTheme = () => {
-  const currentThemeMode = localStorage.getItem("theme");
-  if (currentThemeMode === "system") {
-    document.body.classList.toggle("dark", prefersDark.matches);
-  } else {
-    document.body.classList.toggle("dark", currentThemeMode === "dark");
-  }
-};
-
-toggleDarkTheme();
-
-prefersDark.addEventListener("change", () => toggleDarkTheme());
-
-const selectChanges = (ev: SelectCustomEvent) => {
-  localStorage.setItem("theme", ev.detail.value);
-  toggleDarkTheme();
-};
+onMounted(() => {
```

Comment:
Do we need an onMounted to call updateTheme() here? 

To me it make more sense to call updateTheme() whenever user opens up the App instead of navigate to setting page.

Reply:
True, I have called this from `main.ts` and forgot to remove this one here.